File: ios/Classes/Base/SurfaceViewProtocol.swift
import AgoraRtcKit
import Foundation

protocol SurfaceViewProtocol {
    func setData(_ engine: AgoraRtcEngineKit, _ channel: AgoraRtcChannel?, _ uid: UInt)

    func setRenderMode(_ engine: AgoraRtcEngineKit, _ renderMode: UInt)

    func setMirrorMode(_ engine: AgoraRtcEngineKit, _ mirrorMode: UInt)
}

File: android/src/main/java/io/agora/rtc/base/SurfaceViewProtocol.kt
package io.agora.rtc.base

import android.content.Context
import android.view.SurfaceView
import android.widget.FrameLayout
import io.agora.rtc.RtcChannel
import io.agora.rtc.RtcEngine
import io.agora.rtc.video.VideoCanvas
import java.lang.ref.WeakReference

interface SurfaceViewProtocol {
  fun setZOrderMediaOverlay(isMediaOverlay: Boolean)

  fun setZOrderOnTop(onTop: Boolean)

  fun setData(engine: RtcEngine, channel: RtcChannel?, uid: Number)

  fun setRenderMode(engine: RtcEngine, @Annotations.AgoraVideoRenderMode renderMode: Int)

  fun setMirrorMode(engine: RtcEngine, @Annotations.AgoraVideoMirrorMode mirrorMode: Int)
}
